dist/gcc-compatible/Hacl_Hash_Blake2b.h

```diff
@@ -0,0 +1,142 @@
+/* MIT License
+ *
+ * Copyright (c) 2016-2022 INRIA, CMU and Microsoft Corporation
+ * Copyright (c) 2022-2023 HACL* Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef __Hacl_Hash_Blake2b_H
+#define __Hacl_Hash_Blake2b_H
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+#include <string.h>
+#include "krml/internal/types.h"
+#include "krml/lowstar_endianness.h"
+#include "krml/internal/target.h"
+
+#include "Lib_Memzero0.h"
+#include "Hacl_Krmllib.h"
+
+void Hacl_Hash_Blake2b_init(uint64_t *hash, uint32_t kk, uint32_t nn);
+
+void
+Hacl_Hash_Blake2b_update_key(
+  uint64_t *wv,
```

Comment:
I'm not sure what `wv` is. But the arguments in these stateful functions appear to be in the wrong order.

Reply:
this is another thing that is caused by the original F* type, which uses the Lib helpers -- this could be fixed at the expense of defining other abbreviations, such as `length_of` instead of `lbuffer` -- hopefully this won't break the verification

FYI, `wv` is working vector, but it took me years to understand that... I think we should also fix argument order and names while we're at it